core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -43,6 +45,12 @@ type OutgoingConnectorHandler struct {
 	outgoingRateLimiter *common.RateLimiter
 	responses           *responses
 	selectorOpts        []func(*RoundRobinSelector)
+
+	// toRetryableFunc is a func that may be injected and returns a parameter free
+	// operation that may be retried
+	toRetryableFunc func(context.Context, string, capabilities.Request) func() (*api.Message, error)
```

Comment:
I get that `toRetryableFunc` and `toBackoffOpts` are injectable for easier testing, but is there another way that we can have them testable? Injectable functions have high cognitive load to read. They have a place, but seems overkill here.